language-support/scala/codegen/codegen.bzl

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+def dar_to_scala(**kwargs):
+    """
+    Generate scala code from provided DAR files.
+
+    This macro does not try to compile the generated classes it just outputs the `srcjar` with generated files.
+    """
+    name = kwargs["name"]
+    dars = kwargs["srcs"]
+    package_prefix = kwargs["package_prefix"]
+    verbosity = kwargs.get("verbosity", "2")
+    srcjar_out = kwargs["srcjar_out"]
+
+    src_out = name + "-srcs"
+
+    cmd = """
+        $(execpath //language-support/scala/codegen:codegen-main) --output-directory={gen_out} --verbosity={verbosity} {gen_in}
+        $(execpath @local_jdk//:bin/jar) -cf $@ -C {gen_out} .
```

Comment:
Unfortunately `jar -cf` is not reproducible, which is bad for caching. Bazel includes a `zipper` tool for that purpose. See [`daml/proto.bzl`](https://github.com/digital-asset/daml/blob/44c67547f58ff37f03c797af6cac27f61684e90c/bazel_tools/proto.bzl#L95-L121). They also use it to create a `srcjar`, in that case from proto generated code. To use that it might be easier to express `dar_to_scala` as a proper Bazel rule, rather than a macro. That should be straight forward following the example of `proto_gen`.

Reply:
cc @gerolf-da as this is "borrowed" from java codegen rule